kommunicate/src/main/java/io/kommunicate/async/KmAppSettingTask.java

```diff
@@ -0,0 +1,53 @@
+package io.kommunicate.async;
+
+import android.content.Context;
+import android.os.AsyncTask;
+
+import java.lang.ref.WeakReference;
+
+import io.kommunicate.callbacks.KmCallback;
+import io.kommunicate.models.KmAppSettingModel;
+import io.kommunicate.utils.KmAppSettingPreferences;
+
+public class KmAppSettingTask extends AsyncTask<Void, Void, KmAppSettingModel> {
+
+    private WeakReference<Context> context;
+    private String appId;
+    private String error;
+    private KmCallback callback;
+
+    public KmAppSettingTask(Context context, String appId, KmCallback callback) {
+        this.context = new WeakReference<>(context);
+        this.appId = appId;
+        this.callback = callback;
+    }
+
+    @Override
+    protected KmAppSettingModel doInBackground(Void... voids) {
+        try {
+            KmAppSettingPreferences.getInstance().clearInstance();
+            return KmAppSettingPreferences.fetchAppSetting(context.get(), appId);
```

Comment:
@reytum would it be better if we fetch the settings first and check that the call was successful and then call .clearInstance()? in case if app settings fail?

Reply:
We could do that.

Reply:
Done

Reply:
is it committed? code looks same as before